src/powershell/Public/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,214 @@
+
+    .EXAMPLE
+    .\ConvertTo-FinOpsSchema.ps1 -ActualCost .\Detail_Account_196312_202309_en.csv -outputFilePath .\Detail_Account_196312_202309_en_filtered.csv -ExportAllColumns $true -Force
+
+    # This will convert the input file to the output file using the default column mapping and export all columns. The operation will not run a test and will not prompt the user to continue. 
+
+    .EXAMPLE
+    .\ConvertTo-FinOpsSchema.ps1 -ActualCost .\Detail_Account_196312_202309_en.csv -outputFilePath .\Detail_Account_196312_202309_en_filtered.csv -ExportAllColumns $true -RunTest $true -rows 1000 -columns 10
+    # This will run a test to estimate the processing time and then prompt the user to continue. If the user enters Y or y, the operation will continue. If the user enters N or n, the operation will be aborted. 
+
+    .EXAMPLE
+    .\ConvertTo-FinOpsSchema.ps1 -ActualCost .\Detail_Account_196312_202309_en.csv -outputFilePath .\Detail_Account_196312_202309_en_filtered.csv -ExportAllColumns $false -RunTest $true -rows 1000 -columns 10 -Force
+    # This will run a test to estimate the processing time and then continue without prompting the user. It will export only the columns specified in the column mapping.
+#>
+
+<# Parameter Block #>
+param(
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ActualCost,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $outputFilePath,
+
+    [Parameter(Mandatory = $false)]
+    [bool]
+    $RunTest = $false,
+
+    [Parameter(Mandatory = $false)]
+    [ValidateRange(0, 100000)]
+    [int]
+    $rows = 1000,
+
+    [Parameter(Mandatory = $false)]
+    [ValidateRange(0, 20)]
+    [int]
+    $columns = 10,
+
+    [Parameter(Mandatory = $false)]
+    [bool]
+    $ExportAllColumns = $true,
+
+    [Parameter(Mandatory = $false)]
+    [switch]
+    $Force
+)
+
+# Define the input and output file paths
+$ActualCost=".\Detail_Account_196312_202309_en.csv"
+$outputFilePath=".\Detail_Account_196312_202309_en_filtered.csv"
+
+# Define the column mapping
+# The keys are the original column names, and the values are the new column names. If the value is the same as the key, the column will not be renamed.
+# If the value is not the same as the key, the column will be renamed and the value will be used as the new column name.
+# If the key is not present in the input data, the column will not be included in the output data.
+$columnMapping = @{
+    billingAccountId = 'BillingAccountId'
+    billingAccountName = 'BillingAccountName'
+    ftk_AccountType = 'ftk_AccountType'
+    BillingProfileId = 'BillingProfileId'
+    SubscriptionId = 'SubscriptionId'
+}
+
+<# Helper Functions #>
+function Add-AccountType {
+    [CmdletBinding()]
+    param 
+    (
+        [PSCustomObject]
+        $Object
+    )
+
+    $ftk_AccountType = switch ($true) {
```

Comment:
I like the idea of a switch, when you pass in a value. Unlike the if, the value does not need to evaluate to True or False, it can be any value.
But other than that, I wanted to switch it up and not use if blocks all throughout my code.